src/components/PopupWithForm.js

```diff
@@ -36,14 +36,26 @@ export default class PopupWithForm extends Popup {
       this._submitButton.disabled = true;
     } else {
       this._setButtonText(this._originalButtonText);
-      this._submitButton.disabled = false; 
+      this._submitButton.disabled = false;
     }
   }
 
   _handleSubmit(event) {
     event.preventDefault();
     const inputValues = this._getInputValues();
+    this.renderLoading(true);
     this._handleFormSubmit(inputValues);
+    Promise.resolve(this._handleFormSubmit(inputValues))
+      .then(() => {
+        this._resetForm();
+        this.close();
+      })
+      .catch((err) => {
+        console.error("Error submitting form:", err);
+      })
+      .finally(() => {
+        this.renderLoading(false);
+      });
```

Comment:
Please, delete `then`, `catch`, `finally` from `PopupWithForm` because it they don't work like that. You don't **wait** for the server response here